src/dialogs/alwaysonbotDialogs/alwaysOnBotDialog.ts

```diff
@@ -0,0 +1,104 @@
+
+
+const MAIN_WATERFALL_DIALOG = 'mainWaterfallDialog';
+const CHOICE_PROMPT = 'CHOICE_PROMPT';
+
+export const ALWAYS_ON_BOT_DIALOG_STEP = 'ALWAYS_ON_BOT_DIALOG_STEP';
+
+export class AlwaysOnBotDialog extends ComponentDialog {
+
+    constructor() {
+        super(ALWAYS_ON_BOT_DIALOG_STEP);
+
+        if (!UpdateProfileDialog) throw new Error('[MainDialog]: Missing parameter \'updateProfileDialog\' is required');
+
+        // Define the main dialog and its related components.
+        // This is a sample "book a flight" dialog.
+        this.addDialog(new TextPrompt('TEXT_PROMPT'))
+            .addDialog(new UpdateProfileDialog())
+            .addDialog(new FeedbackDialog())
+            .addDialog(new CommonChoiceCheckDialog())
+            .addDialog(new oASBenefitDialog())
+            .addDialog(new ChoicePrompt(CHOICE_PROMPT,this.CustomChoiceValidator))
+            .addDialog(new WaterfallDialog(MAIN_WATERFALL_DIALOG, [
+                this.introStep.bind(this),
+                this.actStep.bind(this)
+            ]));
+
+        this.initialDialogId = MAIN_WATERFALL_DIALOG;
+    }
+
+    /**
+     * The run method handles the incoming activity (in the form of a DialogContext) and passes it through the dialog system.
+     * If no dialog is active, it will start the default dialog.
+     * @param {TurnContext} context
+     */
+    public async run(context: TurnContext, accessor: StatePropertyAccessor<DialogState>) {
+        const dialogSet = new DialogSet(accessor);
+        dialogSet.add(this);
+        const dialogContext = await dialogSet.createContext(context);
+        const results = await dialogContext.continueDialog();
+        if (results.status === DialogTurnStatus.empty) {
+            await dialogContext.beginDialog(this.id);
+        }
+    }
+
+    /**
+     * First step in the waterfall dialog. Prompts the user for a command.
+     */
+
+    // validates all the prompts
+     private async CustomChoiceValidator(promptContext: PromptValidatorContext<Choice>) {
+        return true;
+    }
+    private async introStep(stepContext: WaterfallStepContext): Promise<DialogTurnResult> {
+        
+       let commonPromptValidatorModel = new CommonPromptValidatorModel(
+        ["IwanttoUpdateMyPersonalInformation", "IhaveaQuestionAboutOASPension"],
+        3,
+        'AlwaysOnBot'
+    );
+    //call dialog
+    return await stepContext.beginDialog(COMMON_CHOICE_CHECK_DIALOG, commonPromptValidatorModel); 
+    }
+    
+    private async actStep(stepContext: WaterfallStepContext): Promise<DialogTurnResult> {
+        const commonPromptValidatorModel = stepContext.result as CommonPromptValidatorModel;
+
+        if (commonPromptValidatorModel != null && commonPromptValidatorModel.status)
+        {
+            switch (commonPromptValidatorModel.result) {
+                case 'IwanttoUpdateMyPersonalInformation':
```

Comment:
the naming is not consistent in these files

in callbackDialog.ts line 47, YesIWantToRequestCall but here it change to  IwanttoUpdateMyPersonalInformation